### ISSUE TITLE
JSON: Field names should account for `r#`

### ISSUE THREAD

**Issue by fosskers** [tokio-rs/tracing#3052]
### Version

`0.3.18`

### Platform

> Linux caddi 6.10.3-arch1-2 #1 SMP PREEMPT_DYNAMIC Tue, 06 Aug 2024 07:21:19 +0000 x86_64 GNU/Linux

### Description

Rust allows us to give struct fields the same name as reserved keywords, such as `type`, _if_ we prefix that name with `r#`, as shown here:

```rust
struct Thing {
    amount: u32,
    r#type: String,
}
```
When serializing this into JSON via `serde`, we see the `r#` ignored in the output:
```rust
#[derive(Debug, serde::Serialize)]
struct Thing {
    amount: u32,
    r#type: String,
}

fn main() {
    let c = Thing {
        amount: 1,
        r#type: "hi".to_string(),
    };

    let json = serde_json::to_string(&c).unwrap();
    println!("{json}");
}
```
> {"amount":1,"type":"hi"}

However, the JSON support within `tracing-subscriber` does not do this, instead rendering the field name as `r#type`.

Suggestion: it should account for any field names that start with `r#` and avoid rendering that piece.

**Comment by mladedav** [comment]
The JSON subscriber uses serde internally so the output should be the same.

Are you maybe using `valuable`? That would change how the serialization works and I've noticed you were using it in other recent issues.

**Comment by fosskers** [comment]
Yes, using `valuable` in this case.

**Comment by mladedav** [comment]
I don't think so, you'd need a copy of the fmt subscriber without the valuable feature.

But in this case I think this should be fixed in `valuable` itself.

**Comment by fosskers** [comment]
So it might not be how the `tracing-subscriber` JSON formatter is _rendering_ the field name, but how `valuable` is reporting the field name to begin with?
